src/dijkstra_spath.jl

```diff
@@ -95,24 +99,30 @@ end
 #
 ###################################################################
 
-function set_source!{V,D}(state::DijkstraStates{V,D}, g::AbstractGraph{V}, s::V)
+function set_source!{V,D}(state::DijkstraStates{V,D}, g::AbstractGraph{V}, s::V; all_paths=false)
     i = vertex_index(s, g)
     state.parents[i] = s
     state.hasparent[i] = true
     state.dists[i] = 0
     state.colormap[i] = 2
+    if all_paths
+        state.pathcounts[i] = 1
+        state.allparents[i] = [s]
+    end
 end
 
 function process_neighbors!{V,D,Heap,H}(
     state::DijkstraStates{V,D,Heap,H},
     graph::AbstractGraph{V},
     edge_dists::AbstractEdgePropertyInspector{D},
-    u::V, du::D, visitor::AbstractDijkstraVisitor)
+    u::V, du::D, visitor::AbstractDijkstraVisitor; all_paths=false)
 
     dists::Vector{D} = state.dists
     parents::Vector{V} = state.parents
     hasparent::Vector{Bool} = state.hasparent
     colormap::Vector{Int} = state.colormap
+    pathcounts::Vector{Int} = state.pathcounts
```

Comment:
Could you wrap these two lines in an if-block? AIUI they will currently cause unnecessary allocation and GC in the `all_paths=false` case, but I haven't tested this.